code/modules/events/disease_outbreak.dm

```diff
@@ -271,25 +279,40 @@
 				/datum/symptom/fire,
 				/datum/symptom/flesh_death,
 				/datum/symptom/flesh_eating,
-				/datum/symptom/narcolepsy,
-				/datum/symptom/voice_change,
+				/datum/symptom/visionloss,
 			)
 
-	for(var/i in 1 to max_symptoms)
+	var/current_severity = 0
+
+	while(symptoms.len < max_symptoms)
 		var/datum/symptom/chosen_symptom = pick_n_take(possible_symptoms)
-		if(chosen_symptom)
-			var/datum/symptom/new_symptom = new chosen_symptom
-			symptoms += new_symptom
 
-	if(prob(50))
-		symptoms += new /datum/symptom/viralevolution
+		if(!chosen_symptom)
+			CRASH("Advanced disease could not pick a symptom!")
 
-	if(prob(50))
-		symptoms += new /datum/symptom/youth
+		//Checks if the chosen symptom is severe enough to meet requested severity. If not, pick a new symptom.
+		//If we've met requested severity already, we don't care and will keep the chosen symptom.
+		var/datum/symptom/new_symptom = new chosen_symptom
 
-	Refresh()
+		if((current_severity < requested_severity) && (new_symptom.severity < requested_severity))
+			continue
+
+		symptoms += new_symptom
 
-	name = "Sample #[rand(1, 9999)]"
+		//Applies the illness name based on the most severe symptom.
+		if(new_symptom.severity > current_severity)
+			name = "[new_symptom.illness]"
+			current_severity = new_symptom.severity
+
+	//Finalize the generated virus with EY on the end.
+	symptoms += new /datum/symptom/youth
+
+	//Advanced variants have increased resistance but spread slower
+	if(prob(40))
+		symptoms += new /datum/symptom/viraladaptation
```

Comment:
Why're we changing it from viralevolution to viraladaptation?

Reply:
I was generating more diseases today working on the other things from the review, and wanted to slow down the disease progression a bit from what I was seeing. So swap out some speed for resistance, and the stage_prob multiplier is reduced from 0.5 to 0.4. This'll make the final product more consistent, less outliers with say, stage speed 12.

Reply:
After sleeping on this, I shouldn't try to box in the values so much, originally I had both Adaptation and Evolution in there, and I think I'll switch it back.